### PR TITLE
fix: rename "Whitelist" to "Allow list"

### PR DESCRIPTION
**Pull request by freesteph**

### CODE REVIEW

File: app/views/super_admin/whitelists/email_domains/_confirm_remove_email_domain.html.erb [hunk 1]
```diff
@@ -3,6 +3,6 @@
     Are you sure you want to remove this email domain?
   </h2>
   <div class="govuk-error-summary__body">
-    <%= button_to "Yes, remove #{params[:domain_to_remove]} from the whitelist", super_admin_whitelist_email_domain_path(params[:id]), method: :delete, class: "govuk-button red-button" %>
+    <%= button_to "Yes, remove #{params[:domain_to_remove]} from the allowed list", super_admin_whitelist_email_domain_path(params[:id]), method: :delete, class: "govuk-button red-button" %>
```
Comment (bethwalton) [inline app/views/super_admin/whitelists/email_domains/_confirm_remove_email_domain.html.erb:6]: Think this should be "allow list"?

File: spec/features/super_admin/whitelists/authorised_email_domain_spec.rb [hunk 2]
```diff
@@ -71,24 +71,24 @@
       end
 
       it "removes a domain" do
-        expect { click_on "Yes, remove #{some_domain} from the whitelist" }.to change(AuthorisedEmailDomain, :count).by(-1)
+        expect { click_on "Yes, remove #{some_domain} from the allowed list" }.to change(AuthorisedEmailDomain, :count).by(-1)
```
Comment (bethwalton) [inline spec/features/super_admin/whitelists/authorised_email_domain_spec.rb]: Can these ones all be 'allow list' too?

**Review by bethwalton — COMMENTED** [review]
Just those changes of "allowed list" to "allow list". 

Everything else looks great, thank you! :D

**Review by alan-gds — CHANGES_REQUESTED** [review]
I've not done a full review yet, but I did a quick search to see if any user-facing text was missed and found a couple of cases; it'd be worth fixing these while I continue to look at the changes that have been made so far.

```
app/models/organisation.rb
25:      errors.add(:base, "#{name} isn't a whitelisted organisation")
```

```
app/views/help/user_support.html.erb
10:    <p class="govuk-body">Contact us for help and support with users connecting to GovWifi in your organisation. We can assist with user log information and whitelisting public sector email domains.</p>
```

**Comment by freesteph** [comment]
> I've not done a full review, but I did a quick search to see if any user-facing text was missed and found a couple of cases.
> 
> ```
> app/models/organisation.rb
> 25:      errors.add(:base, "#{name} isn't a whitelisted organisation")
> ```

Updated, thanks.

> 
> ```
> app/views/help/user_support.html.erb
> 10:    <p class="govuk-body">Contact us for help and support with users connecting to GovWifi in your organisation. We can assist with user log information and whitelisting public sector email domains.</p>
> ```

Discussed with @bethwalton; sentence deemed vague so altogether removed.

:heavy_check_mark:

**Review by bethwalton — APPROVED** [review]

**Review by alan-gds — APPROVED** [review]
Looks good to me.  Would be nice to change internal mentions of whitelist in the code and in various paths at some point (for consistency), but this PR covers the more urgent job of dealing with all the content that's visible on actual pages.

**Pull request merged**